web3/contract.py

```diff
@@ -645,6 +665,103 @@ def _encode_constructor_data(cls, args=None, kwargs=None):
         return deploy_data
 
 
+class ContractConstructor:
+    """
+    Class for contract constructor API.
+    """
+    def __init__(self, web3, address, abi, bytecode, args, kwargs):
+        self.web3 = web3
+        self.address = address
```

Comment:
When is the address going to be helpful in the constructor? Maybe we want to raise an exception if the user tries to issue a deployment transaction for a contract that already has an address?

I don't see any other good way to use the address value with the constructor.

Reply:
True. I will remove it.

Reply:
Thanks! It is removed.